membrane_curvature/lib/mods.py

```diff
@@ -1,89 +1,105 @@
-        Dictionary to store values in every iteration over frames.
-    n_cells : int.
-        number of cells in the grid of size `max_width`.
-    selection : AtomGroup
-        AtomGroup of reference selection to define the surface
-        of the membrane.
-    max_width : int.
-        Maximum width of simulation box.
+    coordinates : tuple.
+        Numpy.ndarray with shape=(n_atoms, 3).
+    n_x_bins : int.
+        Number of bins in grid in the x dimension. 
+    n_y_bins : int.
+        Number of bins in grid in the y dimension. 
+    x_range : tuple
+        Range of indexes in grid in the x dimension with shape=(0, max_width_x).
+    y_range : tuple
+        Range of indexes in grid in the y dimension with shape=(0, max_width_y)
+
 
     Returns
     -------
-
-    Returns a dictionary with tuples of [i,j] as keys, where 0<i<`max_width`,
-    and 0<j<`max_width`. The resulting values for each key are a list of
-    `len=(n_frames)` containing the extracted z_coordinate a given [i,j] cell
-    in each frame.
+    Returns surface derived from z coordinates in grid of `n_x_bins` x `n_y_bins` dimensions.
 
     """
 
-    grid_count_frames = np.zeros([n_cells, n_cells])
+    z_ref = np.zeros((n_x_bins, n_y_bins))
+    grid_z_coordinates = np.zeros((n_x_bins, n_y_bins))
+    grid_norm_unit = np.zeros((n_x_bins, n_y_bins))
+
+    x_factor = n_x_bins / (x_range[1] - x_range[0])
+    y_factor = n_y_bins / (y_range[1] - y_range[0])
 
-    for ts in universe.trajectory:
+    x_coords, y_coords, z_coords = coordinates.T
 
-        grid_z_coordinates = np.zeros([n_cells, n_cells])
-        grid_norm_unit = np.zeros([n_cells, n_cells])
+    cell_x_floor = np.floor(x_coords * x_factor).astype(int)
+    cell_y_floor = np.floor(y_coords * y_factor).astype(int)
 
-        factor = np.float32(n_cells / max_width)
+    for l, m, z in zip(cell_x_floor, cell_y_floor, z_coords):
 
-        for bead in selection:
-            x, y, z = bead.position/10
+        try:
+            grid_z_coordinates[l, m] += z
+            grid_norm_unit[l, m] += 1
 
-            l, m = grid_map((x, y), factor)
+        except IndexError:
+            print("Atom outside grid boundaries. Skipping atom.")
 
-            try:
-                grid_z_coordinates[l, m] += z
-                grid_norm_unit[l, m] += 1
+    z_surface = avg_unit_cell(z_ref, grid_z_coordinates, grid_norm_unit)
 
-            except ValueError:
-                print("Atom outside grid boundaries. Skipping atom {}".format(bead))
+    return z_surface
 
-        for i, j in it.product(range(n_cells), range(n_cells)):
-            if grid_norm_unit[i, j] > 0:
-                z_Ref[i, j] += grid_z_coordinates[i, j] / grid_norm_unit[i, j]
-                grid_count_frames[i, j] += 1
 
-    for i, j in it.product(range(n_cells), range(n_cells)):
-        if grid_count_frames[i, j] > 0:
-            z_Ref[i, j] /= grid_count_frames[i, j]
+def avg_unit_cell(z_ref, grid_z_coordinates, grid_norm_unit):
```

Comment:
Could you come up with a more descriptive name, like `average_z_coordinates`? I'm not really sure what averaging the unit cell means with respect to the arguments given to the function.

Reply:
Sorry, missed this comment. Will be pushed shortly.